server/data_common/data_adaptor.py

```diff
@@ -122,6 +122,12 @@ def get_var_keys(self):
         # return list of keys
         pass
 
+    def __enter__(self):
+        return self
```

Comment:
this was a good little learning for me -- for some reason I always thought that `__enter__` had a default definition that returned self and that `__exit__` had a default definition that did nothing.

But apparently you had to define these, as I just learned from a little test on my own.

TIL 👍

Reply:
yeah, also still adjusting to not having explicit interfaces and/or base classes for all the special methods Python supports.  https://en.wikipedia.org/wiki/Duck_typing